lib/class-wp-rest-batch-controller.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * REST API: WP_REST_Batch_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Core class used to perform abtch requests.
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Batch_Controller {
+
+	/**
+	 * Registers the REST API route.
+	 *
+	 * @since 9.0.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			'__experimental',
+			'batch',
+			array(
+				'callback'            => array( $this, 'serve_batch_request' ),
+				'permission_callback' => '__return_true',
+				'methods'             => array( 'POST', 'PUT', 'PATCH', 'DELETE' ),
+				'args'                => array(
+					'validation' => array(
+						'type'    => 'string',
+						'enum'    => array( 'require-all-validate', 'normal' ),
+						'default' => 'normal',
+					),
+					'requests'   => array(
+						'required' => true,
+						'type'     => 'array',
+						'maxItems' => 25,
```

Comment:
Why 25? I imagine 50 or more requests won't be uncommon when saving menus. We have to draw the line somewhere, but 25 seems a bit low.

Reply:
Totally arbitrary. I definitely think we'll want to fine tune this once we get menus properly working with batching to see realistic performance characteristics.